Src/Infrastructure/CaseStudy.Infrastructure/Repositories/MenuSettingsRepo.cs

```diff
@@ -0,0 +1,126 @@
+﻿using CaseStudy.Core.Contracts.IReposritories;
+using CaseStudy.Core.Models;
+using CaseStudy.Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+using CaseStudy.Application.DTOS;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CaseStudy.Core.DTO;
+
+namespace CaseStudy.Infrastructure.Repositories
+{
+    public class MenuSettingsRepo : IMenuSettingsRepo
+    {
+        private PrjContext context;
+        public MenuSettingsRepo(PrjContext prjContext)
+        {
+            context = prjContext;
+        }
+
+
+        public async Task<IEnumerable<MenuSettings>> GetMenuSettings()
+        {
+            try
+            {
+                return await context.menuSettings.ToListAsync();
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Cannot Fetch MenuSettings", ex);
+
+            }
+        }
+        public async Task<MenuSettings> GetMenuSettingsById(int DealerId)
+        {
+            try
+            {
+
+                var settings= await context.menuSettings.FirstOrDefaultAsync(ms => ms.DealerId == DealerId);
+                
+                return settings;
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Cannot Fetch MenuSettings of the Dealer:- {DealerId}", ex);
+
+            }
+        }
+        public async Task<bool> AddMenuSettings(MenuSettings menuSettings)
+        {
+            try
+            {
+                var Dealer = await context.dealers.FindAsync(menuSettings.DealerId);
+                if(Dealer == null) {
+                 return false;
+                }
+                await context.AddAsync(menuSettings);
+                await context.SaveChangesAsync();
+                return true;
+            }
+            catch
+            {
+                throw new Exception();
+            }
+        }
+
+        public async Task<bool> UpdateMenuSettings(MenuSettingsDTO menuSetting)
+        {
+            try
+            {
+                var existingMenuSetting = await context.menuSettings
+           .FirstOrDefaultAsync(ms => ms.DealerId == menuSetting.DealerId);
+                if (existingMenuSetting == null)
+                {
+                    return false;
+                }
+                if (menuSetting.MenuType != null)
```

Comment:
![image](https://github.com/SSowmya7/CaseStudy/assets/113988510/33e2205a-105d-4cd4-8d5e-c54299197fbd)
no need of if